scene/3d/xr_nodes.cpp

```diff
@@ -428,12 +440,6 @@ PackedStringArray XRNode3D::get_configuration_warnings() const {
 	PackedStringArray warnings = Node::get_configuration_warnings();
 
 	if (is_visible() && is_inside_tree()) {
-		// must be child node of XROrigin!
```

Comment:
Still not 100% convinced there is a valid situation for an `XRNode3D` with tracking data not being a child of `XROrigin3D` but ok..

Reply:
Yeah, I also think this warning should apply to all `XRNode3D`s. I think in my earlier review, I misinterpreted that the check was still there, just moved down a little bit. But now I see that the warning is only in `XRController3D` now?

Reply:
Added the "XRNode3D may not function as expected without an XROrigin3D node as its parent." to XRNode3D so all inherited nodes generate this warning when the parent exists but isn't an XROrigin3D